tsdb/head_wal.go

```diff
@@ -237,6 +237,13 @@ Outer:
 				}
 				if !created {
 					multiRef[walSeries.Ref] = mSeries.ref
+
+					// Mark the walSeries as deleted, so it is kept in subsequent WAL checkpoints.
+					_, last, err := wlog.Segments(h.wal.Dir())
```

Comment:
This seems overkill.  Can we not use the current segment number as the last one where the old series was seen?

Reply:
We're marking the *new* series as deleted, not the old one, so we need the last segment that has samples for the series record we're currently processing. You're right that scanning the directory here each time doesn't make sense, since the answer should always be the same.

Reply:
OK thanks for the correction.  Now I am worried that the code asserts something that is not true: the new series has not been deleted.

Maybe `deleted` is actually `possiblyDeleted`, since at the time we make use of it we also check whether the series is in the head?

Reply:
Moved this out of the loop in 2a60b841ae4f24e232dc4d42820a022e02139f7f.

Reply:
Renamed to `walExpiries` in 188bae868a6379af04c0d75ad0c7c8cbfc28e4ec -- open to alternate naming :)